tokio/src/fs/read_to_string.rs

```diff
@@ -25,6 +25,36 @@ use std::{io, path::Path};
 /// # }
 /// ```
 pub async fn read_to_string(path: impl AsRef<Path>) -> io::Result<String> {
-    let path = path.as_ref().to_owned();
+    let path = path.as_ref();
+
+    #[cfg(all(
+        tokio_unstable,
+        feature = "io-uring",
+        feature = "rt",
+        feature = "fs",
+        target_os = "linux"
+    ))]
+    {
+        use crate::fs::read_uring;
+
+        let handle = crate::runtime::Handle::current();
+        let driver_handle = handle.inner.driver().io();
+        if driver_handle.check_and_init()? {
+            return read_uring(path).await.and_then(|vec| {
+                String::from_utf8(vec).map_err(|_| {
+                    io::Error::new(
+                        io::ErrorKind::InvalidData,
+                        "stream did not contain valid UTF-8",
```

Comment:
The bytes might be invalid or incomplete. Better use Utf8Error's Display impl:
```suggestion
                        format!("{err}"),
```